<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="UR FTS plugin">
  <xacro:macro name="ur_ft_sensor" params="tf_prefix robot_ip update_rate">
    <ros2_control name="ur_force_torque_sensor" type="sensor">
      <hardware>
        <plugin>ur_force_torque_sensor/URFTSensor</plugin>
        <param name="robot_ip">${robot_ip}</param>
        <param name="update_rate">${update_rate}</param>
      </hardware>

      <sensor name="${tf_prefix}ur_force_torque_sensor">
        <state_interface name="force.x"/>
        <state_interface name="force.y"/>
        <state_interface name="force.z"/>
        <state_interface name="torque.x"/>
        <state_interface name="torque.y"/>
        <state_interface name="torque.z"/>
      </sensor>
    </ros2_control>
  </xacro:macro>
</robot>
